### PR TITLE
issue-1100 - Add junits for AggregatingReducer and BulkIngestKeyAggregatingReducer

### PR DESCRIPTION
**Pull request by friedlou**

### CODE REVIEW

**Review by hlgp — APPROVED** [review]

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducerTest.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        PowerMockito.whenNew(Configuration.class).withAnyArguments().thenReturn(conf);
+        PowerMockito.when(conf.iterator()).thenReturn(confMap.entrySet().iterator());
+        
+        PowerMockito.mockStatic(TableConfigurationUtil.class, new Class[0]);
+        PowerMockito.when(TableConfigurationUtil.getTables((Configuration) Mockito.any(Configuration.class))).thenReturn(tables);
+    }
+    
+    @Test
+    public void testColumnToClassMappingOptMap() {
+        // optMap.put("fam1:qual1", AggregatingReducerTest.testCombiner1.class.getName());
```
Comment (javease) [inline warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducerTest.java]: get rid of these three commented code lines

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+import java.util.Random;
+import java.util.Set;
+
+import static datawave.ingest.mapreduce.job.reduce.AggregatingReducer.MILLISPERDAY;
+import static datawave.ingest.mapreduce.job.reduce.AggregatingReducer.USE_AGGREGATOR_PROPERTY;
+import static datawave.ingest.mapreduce.job.reduce.BulkIngestKeyAggregatingReducer.CONTEXT_WRITER_CLASS;
+import static datawave.ingest.mapreduce.job.reduce.BulkIngestKeyAggregatingReducer.CONTEXT_WRITER_OUTPUT_TABLE_COUNTERS;
+import static datawave.ingest.mapreduce.job.reduce.BulkIngestKeyAggregatingReducer.VERBOSE_COUNTERS;
+import static datawave.ingest.mapreduce.job.reduce.BulkIngestKeyAggregatingReducer.VERBOSE_PARTITIONING_COUNTERS;
+import static org.junit.Assert.*;
```
Comment (javease) [inline warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java]: i don't know if we're using coding standards here, but we generally frown on wildcard imports, i.e. you may want to explicitly mention your imports.

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+        PowerMockito.doAnswer(invocation -> {
+            BulkIngestKey k = invocation.getArgument(0);
+            Value v = invocation.getArgument(1);
+            output.put(k, v);
+            return null;
+        }).when(context).write(Mockito.any(BulkIngestKey.class), Mockito.any(Value.class));
+        PowerMockito.when(context.getCounter(IngestOutput.DUPLICATE_KEY)).thenReturn(duplicateKey);
+        
+        reducer = new BulkIngestKeyAggregatingReducer<>();
+        // reducer.setupContextWriter(conf);
```
Comment (javease) [inline warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java]: remove this commented code if it's not needed

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducerTest.java (hunk 1 above)
Comment (friedlou) [reply warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/AggregatingReducerTest.java]: Removed in latest commit.

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java (hunk 2 above)
Comment (friedlou) [reply warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java]: Modified in latest commit.

File: warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java (hunk 3 above)
Comment (friedlou) [reply warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/reduce/BulkIngestKeyAggregatingReducerTest.java]: Removed in latest commit.

**Review by apmoriarty — APPROVED** [review]
LGTM

**Review by hlgp — APPROVED** [review]

**Pull request merged**